### DISCUSSION TITLE
Header image in HTML won't stay in the same position in another computer

### DISCUSSION THREAD

**Discussion by micaelagimenez** [community/community#21483] [Other Feature Feedback, Questions, & Ideas]
<p>Hello, I made this website with html the other day: <a href="https://ghiblimovs.herokuapp.com/" rel="nofollow noopener">https://ghiblimovs.herokuapp.com/</a><br>
The thing is, that even though the header image is centered on my screen, when my friend opened the website in her computer, the image was way more to the left and top side of the website instead. I have tried changing things and I played around with media queries but something isn’t working… Any help would be very appreciated!</p>
<p>This is my html:</p>
<pre><code class="lang-auto">﻿{% load static %}
&lt;!DOCTYPE html&gt;
&lt;html&gt;
&lt;head&gt;
    &lt;title&gt;Ghibli Studio | Movies&lt;/title&gt;
    &lt;link rel="stylesheet" href="{% static 'core/main.css' %}"&gt;
&lt;/head&gt;
&lt;body&gt;
    &lt;div class=" header"&gt;

    &lt;/div&gt;
    &lt;div class="wrap"&gt;
        &lt;form action='/' method="POST"&gt;
            {% csrf_token %}
            &lt;div class="search"&gt;

                &lt;input type="text" name="search" class="searchTerm" placeholder="Type movie name"&gt;
                &lt;link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"&gt;
                &lt;button type="submit" class="searchButton"&gt;

                    &lt;i class="fa fa-search" style="font-size:24px"&gt;&lt;/i&gt;
                &lt;/button&gt;

            &lt;/div&gt;

        &lt;/form&gt;
    &lt;/div&gt;

    {% if movies %}
    {% for movie in movies %}
    &lt;div&gt;
        &lt;ul class="result"&gt;
            &lt;style&gt;
                @import url('https://fonts.googleapis.com/css2?family=Dosis:wght@300&amp;display=swap');
            &lt;/style&gt;
          
            &lt;h4&gt;{{movie.Title}}&lt;/h4&gt;
            &lt;h5 style= "color:lightslategray;"&gt;
                Release date: {{movie.Release_date}} &lt;br&gt;
                Director: {{movie.Director}}
                &lt;br&gt;Producer: {{movie.Producer}}
                &lt;br&gt;
                &lt;br&gt;{{movie.Description}}
            &lt;/h5&gt;
          

        &lt;/ul&gt;
        {% endfor %}

        {% endif %}
    &lt;/div&gt;

&lt;/body&gt;
&lt;/html&gt;
</code></pre>
<p>This is my css:</p>
<pre><code class="lang-auto">body {
    background: #D4E6F1;
    font-family: 'Open Sans', sans-serif;
}
.header {
    background: url(https://upload.wikimedia.org/wikipedia/sco/thumb/c/ca/Studio_Ghibli_logo.svg/640px-Studio_Ghibli_logo.svg.png) no-repeat;
    font-size: 14px;
    width: 640px;
    height: 307px;
    background-size: 70%;
    display: block;
    margin-top: 80px;
    margin-left: auto;
    margin-right: 340px;
}

.search {
    width: 100%;
    margin-bottom: 150px;
    display: flex;
}

.searchTerm {
    width: 100%;
    border: 3px solid #AEB6BF ;
    border-right: none;
    padding: 5px;
    height: 20px;
    margin-top: 110px;
    border-radius: 5px 0 0 5px;
    outline: none;
    color: #AEB6BF ;
}

    .searchTerm:focus {
        color: #808B96;
    }

.searchButton {
    width: 40px;
    height: 36px;
    border: 2px solid #AEB6BF;
    background: #D4E6F1;
    text-align: center;
    color: #AEB6BF;
    margin-top: 110px;
    border-radius: 0 2px 2px 0;
    cursor: pointer;
    font-size: 20px;
}

/*Resize the wrap to see the search bar change!*/
.wrap {
    width: 30%;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}
html {
    font-family: 'Dosis', sans-serif;
    line-height: 1.6;
    color: #666;
}

h4 {
    margin-bottom: 500px;
    margin-top: -50px;
    width: 20%;
    height: 5px;
    text-align: center;
    font-family: 'Dosis', sans-serif;
    padding: 1.5rem 2rem;
    background: url(https://cdna.artstation.com/p/assets/images/images/017/168/420/large/emrullah-cita-ghibli-clouds.jpg?1554902207);
    background-position: center;
    font-size: 1.3rem;
    color: #ffffff;
    text-shadow: #000000;
    position: absolute;
    top: 63%;
    border-radius: 12px 12px 0 0;
    left: 38%;
}
h5 {
    margin-top: -50px;
    background: #F2F4F4;
    width: 20%;
    height: 290px;
    text-align: justify;
    font-family: 'Dosis', sans-serif;
    padding: 1.5rem 2rem;
    font-size: 1rem;
    position: absolute;
    top: 70%;
    left: 38%;
    overflow: hidden;
    box-shadow: 2px 4px 25px rgba(0, 0, 0, .1);
    border-radius: 0 0 12px 12px;

}
</code></pre>

**Comment by SethClydesdale** [comment, marked as answer]
<p>Hello <a class="mention" href="/micaelagimenez">@micaelagimenez</a>,</p>
<p>The problem with absolutely positioned elements, is that, depending on the viewport and parent element, their positions can differ drastically, unless you have a bunch of media queries to correct those positions.</p>
<p>What you’re attempting can be done without relying on position or transform. Here’s a few suggestions:</p>
<p><strong>.header</strong></p>
<ul>
<li>Add <code>center</code> to the <code>background</code> property; this will center the background on all devices.</li>
<li>Change <code>margin-right: 340px;</code> to <code>margin-right: auto;</code>, with left and right set to <code>auto</code>, it’ll center the header on all devices.</li>
</ul>
<p>You could also put the image in the header as an <code>&lt;img&gt;</code> element, then set the <code>max-width</code> on the image to 100% to resize the image on all devices. <code>text-align:center;</code> on .header should center the image.</p>
<p><strong>.wrap</strong></p>
<ul>
<li>Drop the position, top, left, and transform properties.</li>
<li>Add <code>margin:0 auto;</code> to center the wrap.</li>
</ul>
<p><strong>.searchTerm AND .searchButton</strong></p>
<ul>
<li>Drop <code>margin-top: 110px;</code> or lower its value. With the above rules changed, you’ll most likely not need this.</li>
</ul>
<p>Here is the changed CSS:</p>
<pre><code class="lang-css">body {
    background: #D4E6F1;
    font-family: 'Open Sans', sans-serif;
}
.header {
    background: url(https://upload.wikimedia.org/wikipedia/sco/thumb/c/ca/Studio_Ghibli_logo.svg/640px-Studio_Ghibli_logo.svg.png) no-repeat center;
    font-size: 14px;
    width: 640px;
    height: 307px;
    background-size: 70%;
    display: block;
    margin-top: 80px;
    margin-left: auto;
    margin-right: auto;
}

.search {
    width: 100%;
    margin-bottom: 150px;
    display: flex;
}

.searchTerm {
    width: 100%;
    border: 3px solid #AEB6BF ;
    border-right: none;
    padding: 5px;
    height: 20px;
    border-radius: 5px 0 0 5px;
    outline: none;
    color: #AEB6BF ;
}

    .searchTerm:focus {
        color: #808B96;
    }

.searchButton {
    width: 40px;
    height: 36px;
    border: 2px solid #AEB6BF;
    background: #D4E6F1;
    text-align: center;
    color: #AEB6BF;
    border-radius: 0 2px 2px 0;
    cursor: pointer;
    font-size: 20px;
}

/*Resize the wrap to see the search bar change!*/
.wrap {
    width: 30%;
    margin:0 auto;
}

html {
    font-family: 'Dosis', sans-serif;
    line-height: 1.6;
    color: #666;
}



h4 {
    margin-bottom: 500px;
    margin-top: -50px;
    width: 20%;
    height: 5px;
    text-align: center;
    font-family: 'Dosis', sans-serif;
    padding: 1.5rem 2rem;
    background: url(https://cdna.artstation.com/p/assets/images/images/017/168/420/large/emrullah-cita-ghibli-clouds.jpg?1554902207);
    background-position: center;
    font-size: 1.3rem;
    color: #ffffff;
    text-shadow: #000000;
    position: absolute;
    top: 63%;
    border-radius: 12px 12px 0 0;
    left: 38%;
}

h5 {
    margin-top: -50px;
    background: #F2F4F4;
    width: 20%;
    height: 290px;
    text-align: justify;
    font-family: 'Dosis', sans-serif;
    padding: 1.5rem 2rem;
    font-size: 1rem;
    position: absolute;
    top: 70%;
    left: 38%;
    overflow: hidden;
    box-shadow: 2px 4px 25px rgba(0, 0, 0, .1);
    border-radius: 0 0 12px 12px;

}
</code></pre>
<p>The absolute position property is not all bad though, for example, if you have a <code>position:relative;</code> parent element to anchor it in place, it’ll work just fine on all devices. You’ll usually run into display issues when the parent for the absolute element is the <code>&lt;body&gt;</code> element, because the size of the body is based on the viewer’s viewport. In your case, <code>margin:0 auto;</code> was the best method of centering your header and wrap.</p>
<p>For some more centering methods, check out this guide:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <a href="https://css-tricks.com/centering-css-complete-guide/" target="_blank" rel="nofollow noopener" title="04:56PM - 02 September 2014">CSS-Tricks – 2 Sep 14</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:600/300;"><img src="https://i0.wp.com/css-tricks.com/wp-content/uploads/2014/09/guide-centering.png?fit=600%2C99999&amp;ssl=1" class="thumbnail" width="600" height="300"></div>

<h3><a href="https://css-tricks.com/centering-css-complete-guide/" target="_blank" rel="nofollow noopener">Centering in CSS: A Complete Guide | CSS-Tricks</a></h3>

<p>Centering things in CSS is the poster child of CSS complaining. Why does it have to be so hard? They jeer. I think the issue isn't that it's difficult to</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>P.S.<br>
Love Ghibli movies btw</p>

**Comment by micaelagimenez** [comment]
<p>Thank you so much for the detailed answer, it solved all the issues I had! I’ll remember it for the next time I write some css!</p>
<p>P.S.<br>
Love the Genki study resources</p>

**Comment by denskarlet** [comment]
Yeah sure, f*ck off, spamming on github of all places ffs.

Happy holidays everyone<3
